### PR TITLE
Add new device types

### PR DESCRIPTION
**Pull request by dotandl**
`spotifyd` was lacking new device types (such as `automobile`, `gameconsole` etc.) introduced some time ago in `librespot`.

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -150,6 +158,16 @@ pub enum DeviceType {
...
+    GameConsole = 9,
+    CastAudio = 10,
+    CastVideo = 11,
+    Automobile = 12,
+    Smartwatch = 13,
+    Chromebook = 14,
+    UnknownSpotify = 15,
+    CarThing = 16,
+    Observer = 17,
+    HomeThing = 18,
```
Comment (eladyn) [inline src/config.rs:170]: I'm a bit confused, why we need those numbers in the enum (not only in the changes you introduced, but also above). If you have the time, we could maybe get rid of them? But surely not necessary for this PR to be mergeable.

**Review by eladyn — DISMISSED** [review]
Thank you! Some minor comment that you can address, if you want, but otherwise I'd be ready to merge.

File: src/config.rs (hunk 1 above)
Comment (dotandl) [reply src/config.rs:170]: Well, to be honest I wasn't sure why these enum entries have been explicitly assigned a value either. I've done the same to make the code more consistent but now I'm sure this makes no sense (this is the only enum constructed in such way).

I'll push a commit with the values removed in a few minutes.

**Review by eladyn — APPROVED** [review]
Looks good, awesome!

**Pull request merged**